### ISSUE TITLE
Better overrides of tool wear when punching entities

### ISSUE THREAD

**Issue by R3R3R3** [minetest/minetest#9575]
It's a nasty business trying to override the wear on a tool when a punch is performed. I have code in `minetest.register_on_punchplayer` like the following:

```lua
local item = hitter:get_wielded_item()
...
item:add_wear(65535 / 100)
hitter:set_wielded_item(item)
```

In Minetest 5.1, this results in a temporarily visual flicker of the tool having this custom wear applied, and then the change gets clobbered by something at a 'lower' level than user mods. I don't know what's causing this, or if it's intended behaviour of some other part of the software. It doesn't appear to be client prediction.

Anyhow, it would be nice to have a flexible mechanism for controlling tool wear behaviour when punching entities.

---

Just to further document my misadventures: I figured I could override `minetest.get_hit_params`, since `minetest.get_dig_params` is used as a 'dry-run calculator' of wear on node digs, but that was a false assumption.

I've also since worked around the problem by having the tool wear applied on a delay, to undo the effects of the clobbering 'bug' thing.

**Comment by sfan5** [comment]
> In Minetest 5.1, this results in a temporarily visual flicker of the tool having this custom wear applied, and then the change gets clobbered by something at a 'lower' level than user mods. I don't know what's causing this, or if it's intended behaviour of some other part of the software.

No, that's a bug. It was already fixed by e774d8ca2fa5ead9f6d739a2031e6655a22a5b4d though.
